### DISCUSSION TITLE
Any examples for form data (`curl -F`)

### DISCUSSION THREAD

**Discussion by smokeyhallow** [httprb/http#728] [Q&A] [answered]
I can't seem to find out how to do the seemingly simple command:

`curl -X POST -F @image=hello.png https://example.com/upload`

in `http.rb`

Any help would be greatly appreciated!

**Comment by tarcieri** [comment]
See https://github.com/httprb/form_data#usage

**Comment by ixti** [comment, marked as answer]
@tarcieri I think the question was mostly about HTTP.rb itself. :D

@smokeyhallow See: https://github.com/httprb/http/wiki/Passing-Parameters#file-uploads-via-form-data

``` ruby
HTTP.post("https://example.com/upload", form: {
  image: HTTP::FormData::File.new("hello.png")
})
```

**Reply by smokeyhallow** [reply]
Thanks a lot! Appreciate your time!

**Reply by tycooon** [reply]
I guess this will work with just `image: File.new("hello.png")` btw?

**Reply by ixti** [reply]
Correct
